tests/auth_tests/test_forms.py

```diff
@@ -333,6 +334,21 @@ def test_html_autocomplete_attributes(self):
                 )
 
 
+class UserCreationFormTest(TestDataMixin, TestCase):
+    def test_case_insensitive_username(self):
+        data = {
+            "username": "Testclient",
+            "password1": "test123",
+            "password2": "test123",
+        }
+        form = UserCreationForm(data)
+        self.assertFalse(form.is_valid())
+        self.assertEqual(
+            form["username"].errors,
+            [str(User._meta.get_field("username").error_messages["unique"])],
+        )
```

Comment:
I might prefer the actual error message to the `User._meta.get_field("username").error_messages["unique"]` reference.